.github/workflows/preview-env-build.yml

```diff
@@ -0,0 +1,71 @@
+---
+name: Preview Environment Build
+
+on:
+  pull_request:
+    types: [ labeled,synchronize ]
+
+concurrency:
+  cancel-in-progress: true
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref_name }}
```

Comment:
What should this specific `concurrency` setting achieve for this workflow, can you describe it and put a comment into the file for future readers? [example](https://github.com/camunda/camunda/blob/67f8fbf3948880dd580c1b1fc51b3943afd7f839/.github/workflows/operate-a11y.yml#L16-L17)